core/templates/dev/head/components/HintAndSolutionButtonsDirective.ts

```diff
@@ -21,8 +21,10 @@ oppia.directive('hintAndSolutionButtons', [
     return {
       restrict: 'E',
       scope: {},
+      bindToController: true,
       templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
         '/components/hint_and_solution_buttons_directive.html'),
+      controllerAs: 'hintAndSolutionButtonsCtrl',
       controller: [
         '$scope', '$rootScope', 'HintsAndSolutionManagerService',
```

Comment:
Here and elsewhere, should we remove $scope from the dependency list?

Reply:
There are some directives where we have `$scope.$on` / `$scope.$apply`. I am not able to find a substitute for this for controller binding. So, this part was left the same. It works correctly but we cannot use $scope in Angular2. So, if we don't get a substitute in AngularJS, the other option will be to make these updates when we are just going to make the final upgrade step. What do you suggest here?
\cc @YashJipkate

Reply:
Ah ok, thanks for the explanation! I am fine with leaving it.